platform/cli/cli.ts

```diff
@@ -52,4 +53,24 @@ const run = (argv: { [key: string]: any }) => {
   }
 };
 
+export const runWithLegacyKbnServer = (kbnServer: any) => {
```

Comment:
Btw, if we remove the `process.on` stuff below, I think we should extract this into a separate file (and a separate folder) as there's no cli related stuff left in this function.

Reply:
I agree if we don't have `process.on` stuff it will reside in `legacy` sub-folder.

Reply:
@kjbekkelund regarding `SIGINT` and `SIGTERM`: it seems old platform doesn't handle these events (it only cleans `pid` file on `SIGINT`). I'm not sure if we should handle these events in the new platform __if__ it's run within old one and basically acts as an advanced HTTP server, what do you think?

Reply:
I don't think we should listen for them specifically, but we probably want to listen for something that signals that we should shut down (just to make sure we run all the shutdown stuff correctly for later when we remove the old platform). However, I'm not sure how we indicate that in current platform.